include/deal.II/grid/grid_generator.h

```diff
@@ -292,6 +292,20 @@ namespace GridGenerator
           const std::vector<unsigned int> &holes);
 
   /**
+   * A general quadrilateral in 2d or a general hexahedron in 3d. It is the
+   * responsability of the user to provide the vertices in the right order (see
+   * the documentation of the GeometryInfo class).
+   *
+   * @author Bruno Turcksin
+   * @date 2016
+   */
+  template <int dim>
+  void
+  hyper_quadrilateral(Triangulation<dim> &tria,
```

Comment:
I don't like the name, some ideas: ``single_cell``, ``general_cell``, ``quad_or_hex``